File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashset = set(nums)
        seq = 0

        for num in hashset:
            if (num - 1) not in hashset:
                length = 1
                while (num + length) in hashset:
                    length += 1
                seq = max(length, seq)
                
        return seq

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let count = {};
        let freq = Array.from({ length: nums.length + 1 }, () => []);

        for (let num of nums){
            count[num] = 1 + (count[num] || 0);
        }

        for (let i in count){
            freq[count[i]].push(Number(i));
        }

        let ans = [];

        for(let i = freq.length-1; i > 0; i--){
            for(let n of freq[i]){
                ans.push(n);
                if(ans.length === k){
                    return ans;
                }
            }
        }
    }
}
